packages/babel-helpers/src/helpers.js

```diff
@@ -860,10 +860,27 @@ helpers.arrayWithHoles = helper("7.0.0-beta.0")`
 
 helpers.iterableToArray = helper("7.0.0-beta.0")`
   export default function _iterableToArray(iter) {
-    if (
-      Symbol.iterator in Object(iter) ||
-      Object.prototype.toString.call(iter) === "[object Arguments]"
-    ) return Array.from(iter);
+    // ES6 prototypes that use @@iterator
+    const prototypeWhiteList = [
+      String.prototype,
+      Array.prototype,
+      // Map.prototype,
+      // Set.prototype,
+    ];
+    const iterObject = Object(iter);
+    if (prototypeWhiteList.some(prototype => prototype.isPrototypeOf(iterObject))) {
+      return Array.from(iter);
+    }
+
+    // IsArguments
+    if (iterObject.toString() === "[object Arguments]") {
+      return Array.from(iter);
+    }
+
+    // Check for custom iterator outside ES6 spec, iterable WeakMap, or iterable TypedArray
+    if (Symbol.iterator in iterObject) {
```

Comment:
Wasn't the point of this PR to do a `typeof Symbol !== 'undefined'` check?

Reply:
No. It's to fix the spread regression specifically.

Reply:
But none of your changes will help spreading the `HTMLCollection` from the [issue](https://github.com/babel/babel/issues/9277): `[...document.getElementsByTagName('div')]`. I don't understand what you're looking for then.

Reply:
The input code shows the error, but doesn't accurately detail the regession. It's in the description, which is more accurate. Basically, all `spread` functions fail on IE11 because it's checking for Symbol before checking the prototype. It's a regression from v6. I believe it's caused by https://github.com/babel/babel/commit/4da3f3bc64841e0b4b7906c6c26af73920264095

Reply:
The code in the OP goes through `arrayWithHoles`'s `Array.isArray` check, so this change does nothing still. I need an explicit "this fails in IE 11, doesn't fail with this change" case so that I can review the PR.

Reply:
I think you're missing the point of the `[object Arguments]` check.  That block never executes on environments without `Symbol`. Therefore any function with spread arguments would fail. 

````
function func1(a, b, c) {
  console.log(arguments[0]);
  // expected output: 1

  console.log(arguments[1]);
  // expected output: 2

  console.log(arguments[2]);
  // expected output: 3
}

func1(1, 2, 3);
````

Reply:
Looking back at the issue from 3 months ago, the request was a partial fix for the regression while the more complicated issue of completely eliminating Symbol is explored. That PR has been stuck for a couple of months now.

I think that's why I had the mindset of swapping the order of execution, at least to minimize the issue.

Reply:
So, this is the kind of thing you're trying to fix? https://babeljs.io/repl/build/master#?babili=false&browsers=ie%20%3E%2010&build=&builtIns=false&spec=false&loose=false&code_lz=CYUwNiAuIAQO4EsB2wD2cB0BlAngWwCNUwBuAKADMBXJAY0gVSRgGcq8AKADwBoYc-ALwCUMAN5kYMAE5Qq05lxgBqfipiDyAX0o16jZtBaQOoiVIBuAQ2kwbAcxYwAvHen32IJJBbkpsyHlmNk4MMIcWYW0yMlomFmIQDDBUew4jEwBGPgAmPgBmYSigA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.4.3&externalPlugins=

A executable test case we can add is:

```js
function sum(x, y, z) {
  return x + y + z;
}
function test() {
  var args = arguments;
  return sum(...args);
}

assert.equal(test(1, 2, 3), 6);
```

All that's necessary for this was to change the order of the `||` expression in the old code.

Reply:
Yes, and wrap them in parenthesis. Also, any object that is a prototype of `Array` and `String` would align with the ES6 spec. 

I'm not seeing much more we can do that wouldn't trigger extra corejs2 imports. I have no way of checking for the internal `TypedArray` type. Also, `NodeList`, `HTMLCollection` and others aren't part of the ES6 spec. I realized that this function, at least as it's called, it's checking for array-likes. It's checking for iterables. Therefore, something like this:

````
[...({ 0: 'a', 1: 'b', 2: 'c', length: 3})]
````

would throw an error on an ES6 supporting browser, and therefore should throw an error on Babel. Support for array-likes would have to be `loose`. I think this the best we can do without violating spec.

Reply:
We can add a string check, that's fine. The array check is already covered by `arrayWithHoles` (the difference between `Array.prototype` passing `Array.isArray` and not `Array.prototype.isPrototypeOf(Array.prototype)` doesn't matter). Let's do the string check, then arguments check, then isIterable check.